flask_dance/contrib/salesforce.py

```diff
@@ -68,9 +72,15 @@ def make_salesforce_blueprint(
         client_id=client_id,
         client_secret=client_secret,
         scope=scope,
-        base_url="https://login.salesforce.com/services/oauth2/",
-        authorization_url="https://login.salesforce.com/services/oauth2/authorize",
-        token_url="https://login.salesforce.com/services/oauth2/token",
+        base_url="https://{domain}.salesforce.com/".format(domain=domain),
```

Comment:
Thanks for mentioning this in the changelog. This _seems_ like a more reasonable base URL to me, but can you verify that API calls to Salesforce all generally start with this URL? Just want to make sure that this is the correct value, since the original value was apparently wrong.

Reply:
You can make an authorization process in two ways:
1) Using your instance URL e.g `example.my.salesforce.com/services/oauth2/authorize`
2) Without any information about the instance url: `login.salesforce.com/services/oauth2/authorize` (production), `test.salesforce.com/services/oauth2/authorize` (sandbox)

When it comes to API calls you need to know your instance URL and make calls using that url e.g. `example.my.salesforce.com/services/data/...`

At that moment when the user specifies his domain as `example.my` he can make API calls using url from the previous example. Otherwise, the default URL will be used (login.salesforce.com), and he cannot do that.

I don't want to make the user to specifies his instance URL obligatory because he can be authorized using defaults endpoints and later fetch his instance url.

If you have any other ideas I am open to discuss that :)

Reply:
`base_url` is not used for the authorization process, but for making API calls _after_ the OAuth token is obtained. For example, with the GitHub blueprint, you can run `github.get("/user")`, and it will access the URL `https://api.github.com/user`, because the `base_url` is `https://api.github.com/`.

Can you point me to some documentation for how to make API calls to Salesforce, _after_ you've already gotten an OAuth token? That will help to determine the correct value for `base_url`.

Reply:
Yea I know but to make API calls you need to know your instance URL, but for authorization you don't. So I can leave the base_url as None if the user hasn't provided domain. Sample calls: https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/quickstart_code.htm